### PR TITLE
feat(button): add link button

### PR DESCRIPTION
**Pull request by wangyewei**
Enhanch Button Component:

- Added the `to` attribute.
- Updated the relevant documentation accordingly.
- Added the relevant test units
image

### CODE REVIEW

File: utils/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const getReferenceDataType = (val: unknown): string => Object.prototype.toString.call(val)
+
+export const isString = (val: unknown): val is string => typeof val === 'string'
+export const isNumber = (val: unknown): val is number => typeof val === 'number'
```
Comment (baiwusanyu-c) [inline utils/src/utils.ts:4]: `isString`、`isNumber`、`isArray`、`isObject` 
I extracted part of the commonly used functions to the warehouse https://github.com/baiwusanyu-c/bwsy-utils
`import { isString, isNumbe, isArray, isObject  } from 'baiwusanyu-utils'`

File: utils/src/index.ts [hunk 2]
```diff
@@ -1 +1,3 @@
-export * from './types';
+export * from './types'
+
+export { createCls } from './classnames'
```
Comment (baiwusanyu-c) [inline utils/src/index.ts:3]: Suggest `class-names.ts`

File: preset/src/shortcuts/button.ts [hunk 3]
```diff
@@ -6,37 +6,37 @@ export const buttonShortcuts: Record<string, string> = {
 	'k-button--circle': '!rounded-full p-2',
 
 	'k-button--primary':
-		'bg-ikun-primary-100 text-ikun-primary border-ikun-primary-100',
+		'bg-ikun-primary-100 !text-ikun-primary border-ikun-primary-100',
```
Comment (baiwusanyu-c) [inline preset/src/shortcuts/button.ts:9]: Why use `important`?

File: utils/src/classnames.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { isString, isNumber, isArray, isObject, isRewritten } from "./utils"
+
+export function createCls(
+  ...args:
+    (string | number | Record<string, any> | any[]
+    )[]
+): string | undefined {
```
Comment (baiwusanyu-c) [inline utils/src/classnames.ts:7]: Looks great👍 ！ maybe refactor into all components

**Review by baiwusanyu-c — CHANGES_REQUESTED** [review]

File: preset/src/shortcuts/button.ts (hunk 3 above)
Comment (wangyewei) [reply preset/src/shortcuts/button.ts:9]: I noticed that the styles are being overridden in VitePress when the button tag is set to 'a'.

image

File: utils/src/utils.ts (hunk 1 above)
Comment (wangyewei) [reply utils/src/utils.ts:4]: I think it would be better to add the type predicate `is` in your functions.

**Comment by baiwusanyu-c** [comment]
You deleted other people's pr code, after restoring and synchronizing the latest main branch code, I will merge the pr again

**Review by baiwusanyu-c — APPROVED** [review]

**Pull request merged**